nightingale_dispatcher/src/nightingale_dispatcher/mission_planner.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import rospy
+import actionlib
+import queue
+
+from actionlib_msgs.msg import GoalStatus
+from nightingale_msgs.msg import MissionPlanAction
+from nightingale_dispatcher.handoff_task import HandoffTask
+from nightingale_dispatcher.idle_task import IdleTask
+from nightingale_dispatcher.navigate_task import NavigateTask
+from nightingale_dispatcher.stock_task import StockTask
+from nightingale_dispatcher.task import Task
+from nightingale_dispatcher.triage_task import TriageTask
+
+class MissionPlanner:
+    def __init__(self):
+        rospy.init_node("mission_planner_node")
+
+        self.server = actionlib.SimpleActionServer(
+            "mission_planner", MissionPlanAction, self.goal_cb, False
+        )
+        self.server.start()
+
+        self.handoff_task = HandoffTask()
+        self.idle_task = IdleTask()
+        self.navigate_task = NavigateTask()
+        self.stock_task = StockTask()
+        self.triage_task = TriageTask()
+
+        self.states = queue.Queue()
```

Comment:
I don't agree with using states to describe the functions here. Maybe something like "phases" would be better since its a verb or something the robot is doing. I think state should be reserved as a parameter to describe the robot on a more granular level like: "driving" or "extending arm"